pkg/network/node/egressip.go

```diff
@@ -52,9 +53,11 @@ func newEgressIPWatcher(oc *ovsController, localIP string, masqueradeBit *int32)
 func (eip *egressIPWatcher) Start(networkInformers networkinformers.SharedInformerFactory, iptables *NodeIPTables) error {
 	eip.iptables = iptables
 
-	updates := make(chan *egressVXLANNode, 300)
-	eip.vxlanMonitor = newEgressVXLANMonitor(eip.oc.ovs, eip.tracker, updates)
-	go eip.watchVXLAN(updates)
+	updates := make(chan struct{}, 1)
+	updateNodes := make(map[string]*egressVXLANNode)
+	var updateNodesMux sync.Mutex
+	eip.vxlanMonitor = newEgressVXLANMonitor(eip.oc.ovs, eip.tracker, updates, &updateNodes, &updateNodesMux)
```

Comment:
This is kind of a wacky API. You should just have evm maintain the map by itself, and when eip needs to know, it can call `evm.GetUpdates()` which will lock the existing evm mutex, grab the current map and replace it with an empty one, and return the old map to eip.